Week-3/assignment-3.js

```diff
@@ -0,0 +1,27 @@
+console.log('--------------------');
+console.log('This is assignment 3');
+
+function createIncrement()
+{
+    let count = 0;
+    function increment()
+    { count++;}
+
+    let message = `Count is ${count}`;
+    function log()
+    {
+        console.log(message);
+    }
+    return [increment, log];
+}
+const [increment, log] = createIncrement();
+increment();
+increment();
+increment();
+log();
+
+// Explanation of the function
+
+// Here the increment function is called thrice, so the count will be incremnted by 3 and finally the count value will be 3.
+
+// But the console.log value will be 0 because variable `message` is a not incremented and assigned a value of 0 at the start.
```

Comment:
@soumya213 message is not incremented (statement seems wrong). You can mention that the message is initialised only once during parent function call and its not getting reset, hence the value is 0

Reply:
@RajeshSivanesan Modified the explanation of the message to: 

/*
The increment function is called thrice, so the count will be incremented by 3 and finally the count value will be 3.
But the console.log value for message is initialized only once during parent function call and its not getting reset, hence the value is 0.
*/